pkg/cache/scheduler/tas_flavor_snapshot.go

```diff
@@ -574,7 +588,8 @@ func (s *TASFlavorSnapshot) requiredReplacementDomain(tr *TASPodSetRequests, ta
 	nodeDomain := ta.Domains[0].Values[0]
```

Comment:
I think we need to be smarter about it when it comes to slices, as contrary to regular `required` TAS request where all domains belong to the same top domain, that's not the case for slices. We need to find out which slices is missing a node and pick it. In case no slice is missing any node, it means that the slice consisted of only node so we need to pick whichever, analogously to regular hot swap. Create a UT for that please